packages/react/src/host/RemoteComponent.tsx

```diff
@@ -14,6 +14,7 @@ interface Props {
   receiver: RemoteReceiver;
   component: Serialized<RemoteComponentDescription<any, any>>;
   controller: Controller;
+  __type__?: React.ReactChild;
```

Comment:
This is actually a `ReactComponentType` I think?

Reply:
I'd also love a comment here that explains why it exists

Reply:
Good catch. Added!

Reply:
With this fix do we need still need the casting in the other files?

Reply:
Yes, it does unless we fix the type of the controller too `packages/react/src/host/controller.tsx` but I imagine it would cause more changes than help.